File: python/monoidal.py
class MonoidalCategory:
    def id(self, obj):
        return obj

    def compose(self, f, g):
        return lambda x: f(g(x))

    def tensor(self, a, b):
        return (a, b)

    def tensorMor(self, f, g):
        return lambda x: (f(x[0]), g(x[1]))

File: haskell/lib/Monoidal.hs
class MonoidalCategory obj mor | obj -> mor where
  id :: obj a -> mor a a
  compose :: mor b c -> mor a b -> mor a c
  tensor :: obj a -> obj b -> obj (a, b)
  tensorMor :: mor a a' -> mor b b' -> mor (a, b) (a', b')